internal/eval/store_eval.go

```diff
@@ -473,6 +473,54 @@ func evalZADD(args []string, store *dstore.Store) *EvalResponse {
 	}
 }
 
+// The ZCOUNT command in DiceDB counts the number of members in a sorted set at the specified key
+// whose scores fall within a given range. The command takes three arguments: the key of the sorted set
+// the minimum score, and the maximum score.
+func evalZCOUNT(args []string, store *dstore.Store) *EvalResponse {
+	if len(args) != 3 {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongArgumentCount("ZCOUNT"),
+		}
+	}
+
+	key := args[0]
+	obj := store.Get(key)
+
+	if obj == nil {
+		return &EvalResponse{
+			Result: 0,
+			Error:  nil,
+		}
+	}
+
+	var sortedSet *sortedset.Set
+	var err []byte
+	sortedSet, err = sortedset.FromObject(obj)
+	if err != nil {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongTypeOperation,
+		}
+	}
+	minValue, errMin := utils.ParseScore(args[1])
```

Comment:
Can we try using `strconv.ParseFloat(var, 64)`  directly here instead? ZADD parses the score this way. Can you please verify if we are handling -inf/inf in ZADD?

Thanks

Reply:
As per redis ZADD command should be valid numeric value and same case for us as well, However, Redis range operations such as  ZCOUNT use -inf and +inf to denote unbounded ranges. so i kept it like this. also i think we can keep parseScore as utility function because in future if we are planning to implement similar redis commands like ZREMRANGEBYSCORE,  ZRANGEBYSCORE we can use this utility,  Let me know if we still want to remove it. Also please if we are adding support of this command assign those to me please

Reply:
Please refer to this doc: https://pkg.go.dev/strconv#ParseFloat. I think it can parse +inf and -inf.. This way we remove extra code.

Reply:
Resolved

Reply:
@apoorvyadav1111 resolved this also run the test cases for all the http, resp, websocket all are passing